include/crocoddyl/multibody/actions/constraint-fwddyn.hxx

```diff
@@ -0,0 +1,196 @@
+  boost::shared_ptr<DifferentialActionDataConstraintFwdDynamics> d =
+      boost::dynamic_pointer_cast<DifferentialActionDataConstraintFwdDynamics>(data);
+  if (d != NULL) {
+    return true;
+  } else {
+    return false;
+  }
+}
+
+/*
+template <typename Scalar>
+void DifferentialActionModelConstraintFwdDynamicsTpl<Scalar>::quasiStatic(const
+boost::shared_ptr<DifferentialActionDataAbstract>& data, Eigen::Ref<VectorXs> u, const Eigen::Ref<const VectorXs>& x,
+                                                                     const std::size_t& maxiter, const Scalar& tol) {
+if (static_cast<std::size_t>(u.size()) != nu_) {
+  throw_pretty("Invalid argument: "
+               << "u has wrong dimension (it should be " + std::to_string(nu_) + ")");
+}
+if (static_cast<std::size_t>(x.size()) != state_->get_nx()) {
+  throw_pretty("Invalid argument: "
+               << "x has wrong dimension (it should be " + std::to_string(state_->get_nx()) + ")");
+}
+// Static casting the data
+DifferentialActionDataConstraintFwdDynamicsTpl<Scalar>* d =
+    static_cast<DifferentialActionDataConstraintFwdDynamicsTpl<Scalar>*>(data.get());
+const Eigen::VectorBlock<const Eigen::Ref<const VectorXs>, Eigen::Dynamic> q = x.head(state_->get_nq());
+const Eigen::VectorBlock<const Eigen::Ref<const VectorXs>, Eigen::Dynamic> v = x.tail(state_->get_nv());
+
+d->multibody.actuation->tau = pinocchio::rnea(pinocchio_, d->pinocchio, q, v, VectorXs::Zero(state_->get_nv()));
+actuation_->get_actuated(d->multibody.actuation, u);
+}
+*/
+
+template <typename Scalar>
+pinocchio::ModelTpl<Scalar>& DifferentialActionModelConstraintFwdDynamicsTpl<Scalar>::get_pinocchio() const {
+  return pinocchio_;
+}
+
+template <typename Scalar>
+const boost::shared_ptr<ActuationModelAbstractTpl<Scalar> >&
+DifferentialActionModelConstraintFwdDynamicsTpl<Scalar>::get_actuation() const {
+  return actuation_;
+}
+
+template <typename Scalar>
+const std::vector<pinocchio::RigidConstraintModelTpl<Scalar, 0>,
+                  Eigen::aligned_allocator<pinocchio::RigidConstraintModelTpl<Scalar, 0> > >&
+DifferentialActionModelConstraintFwdDynamicsTpl<Scalar>::get_contacts() const {
+  return contacts_;
+}
+
+template <typename Scalar>
+const boost::shared_ptr<CostModelSumTpl<Scalar> >& DifferentialActionModelConstraintFwdDynamicsTpl<Scalar>::get_costs()
+    const {
+  return costs_;
+}
+
+template <typename Scalar>
+const typename MathBaseTpl<Scalar>::VectorXs& DifferentialActionModelConstraintFwdDynamicsTpl<Scalar>::get_armature()
+    const {
+  return pinocchio_.armature;
+}
+
+template <typename Scalar>
+void DifferentialActionModelConstraintFwdDynamicsTpl<Scalar>::set_armature(const VectorXs& armature) {
+  if (static_cast<std::size_t>(armature.size()) != state_->get_nv()) {
+    throw_pretty("Invalid argument: "
+                 << "The armature dimension is wrong (it should be " + std::to_string(state_->get_nv()) + ")");
+  }
+  pinocchio_.armature = armature;
+}
```

Comment:
Let's remove this function  (also `get_armature`) as this function can directly obtain from `get_pinocchio`.
This avoid code redundancy.